var/spack/repos/builtin/packages/netpbm/package.py

```diff
@@ -170,6 +170,15 @@ def edit(self, spec, prefix):
 
     def build(self, spec, prefix):
         make()
+        if self.run_tests:
+            # Don't run the default command 'make check' for test
+            self.build_time_test_callbacks = []
+
+    @run_after('build')
+    @on_package_attributes(run_tests=True)
+    def make_check_tree(self):
+        # Run custom test command 'make check-tree'
+        make('check-tree')
```

Comment:
FWIW. The tests fail for me:

```
==> Using cached archive: /usr/WS1/dahlgren/releases/spack/var/spack/cache/_source-cache/archive/62/628dbe8490bc43557813d1fedb2720dfdca0b80dd3f2364cb2a45c6ff04b0f18.tgz
==> No patches needed for netpbm
==> netpbm: Executing phase: 'edit'
==> netpbm: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16' 'check-tree'

35 errors found in build log:
  >> 5       /tmp/dahlgren/spack-stage/spack-stage-netpbm-10.73.35-hv5jo3lbo3tg
             vricp5gxkyf7rtiuegce/spack-src/common.mk:643: depend.mk: No such f
             ile or directory
     6       cat /dev/null >depend.mk
...
     4976    make[4]: Entering directory `/tmp/dahlgren/spack-stage/spack-stage
             -netpbm-10.73.35-hv5jo3lbo3tgvricp5gxkyf7rtiuegce/spack-src/conver
             ter/other/fiasco/lib'
     4977    mkdir -p importinc/netpbm

     ...

     6634    FAILURE 4
     6635    NOT TESTABLE 1
     6636    TOTAL TESTABLE 114
     6637    ==================
     6638    All tests done.
     6639    Mon, 13 Jun 2022 18:06:10 +0000
  >> 6640    make: *** [check-tree] Error 1
  
  
```


Are you saying the following runs `make make_check_tree` when you build with `--test=root`?

```suggestion

    def make_check_tree(self):
        # Run custom test command 'make check-tree'
        make('check-tree')
        
    build_time_test_callbacks = ['make_check_tree']
    run_after('build')(PackageBase._run_default_build_time_test_callbacks)
```

after importing `PackageBase` and `run_after` of course:

```
from spack.package_base import PackageBase, run_after
```

FWIW I pretty much get the same output wrt the failure with the latter; however, I don't get the nice failure summary.